File: ui/src/app/shared/models/comment.ts
export class Comment {
    commentId: number;
    postId: number;
    username: string;
    content: string;
    date: Date;
    constructor(param: {
        commentId: number,
        postId: number,
        username: string,
        content: string,
        date: Date
    }) {
        this.commentId = param.commentId;
        this.postId = param.postId;
        this.username = param.username;
        this.content = param.content;
        this.date = param.date;
    }
}

File: src/main/java/com/net/stackthread/entities/Comment.java
package com.net.stackthread.entities;


import lombok.Data;

import javax.persistence.*;
import java.util.Date;


@Entity
@Data
@Table(name = "comments")
public class Comment
{
   @Id
   @GeneratedValue(strategy = GenerationType.IDENTITY)
   private Long id;

   @ManyToOne(fetch = FetchType.LAZY)
   @JoinColumn(name = "post_id", nullable = false)
   private Post post;

   @Lob
   @Column(nullable = false, columnDefinition = "CLOB")
   private String content;

   @Column(nullable = false)
   private Date createdAt = new Date();
}
